In [1]:
import pandas as pd

df = pd.read_csv("bank_churn.csv")
print(df.shape)
print(df[["Geography", "Exited"]].head())

(10000, 18)
  Geography  Exited
0    France       1
1     Spain       0
2    France       1
3    France       0
4     Spain       0


In [2]:
churn_by_country = df.groupby("Geography")["Exited"].mean() * 100
churn_by_country = churn_by_country.reset_index()
churn_by_country.columns = ["Country", "Churn Rate (%)"]
churn_by_country = churn_by_country.sort_values("Churn Rate (%)", ascending=False)
print(churn_by_country)

   Country  Churn Rate (%)
1  Germany       32.443204
2    Spain       16.673395
0   France       16.174711


In [3]:
churn_summary = churn_by_country.copy()
churn_summary["Churn Rate (%)"] = churn_summary["Churn Rate (%)"].round(2)
churn_summary.to_csv("churn_summary.csv", index=False)
print("Saved! Here is your summary:")
print(churn_summary)

Saved! Here is your summary:
   Country  Churn Rate (%)
1  Germany           32.44
2    Spain           16.67
0   France           16.17


# Bank Customer Churn Analysis

## Objective
Analyze 10,000+ bank customer records to identify which country 
has the highest customer churn rate.

## Tools Used
- Python (pandas) — data loading and analysis
- SQL (SQLite) — querying and validating results
- Power BI — data visualization

## Key Finding
Germany has the highest churn rate at 32.44%, which is nearly 
2x higher than France (16.17%) and Spain (16.67%).

## Business Insight
The bank should investigate why German customers are leaving 
at twice the rate of other regions. Possible factors could 
include pricing, competition, or customer service differences 
in that market.


In [4]:
high_balance = {
    "Country": ["Germany", "France", "Spain"],
    "High_Balance_Count": [1968, 1908, 927],
    "Churn_Rate_Pct": [35.87, 17.98, 17.58]
}

df_high_balance = pd.DataFrame(high_balance)
df_high_balance.to_csv("high_balance_churn.csv", index=False)
print(df_high_balance)

   Country  High_Balance_Count  Churn_Rate_Pct
0  Germany                1968           35.87
1   France                1908           17.98
2    Spain                 927           17.58


In [5]:
import pandas as pd

churn_summary = pd.read_csv("churn_summary.csv")
high_balance = pd.read_csv("high_balance_churn.csv")

with pd.ExcelWriter("churn_analysis.xlsx", engine="openpyxl") as writer:
    churn_summary.to_excel(writer, sheet_name="Churn by Country", index=False)
    high_balance.to_excel(writer, sheet_name="High Balance Churn", index=False)

print("Excel file saved!")

Excel file saved!


In [6]:
df["Age Group"] = pd.cut(df["Age"], bins=[0,30,40,50,60,100], 
                          labels=["<30","30-40","40-50","50-60","60+"])
age_churn = df.groupby("Age Group")["Exited"].mean() * 100
age_churn = age_churn.reset_index()
age_churn.columns = ["Age Group", "Churn Rate (%)"]
age_churn["Churn Rate (%)"] = age_churn["Churn Rate (%)"].round(2)
age_churn.to_csv("age_churn.csv", index=False)
print(age_churn)

  Age Group  Churn Rate (%)
0       <30            7.52
1     30-40           12.11
2     40-50           33.97
3     50-60           56.21
4       60+           24.78


In [7]:
gender_churn = df.groupby("Gender")["Exited"].mean() * 100
gender_churn = gender_churn.reset_index()
gender_churn.columns = ["Gender", "Churn Rate (%)"]
gender_churn["Churn Rate (%)"] = gender_churn["Churn Rate (%)"].round(2)
gender_churn.to_csv("gender_churn.csv", index=False)
print(gender_churn)

   Gender  Churn Rate (%)
0  Female           25.07
1    Male           16.47


In [ ]:
with pd.ExcelWriter("churn_analysis.xlsx", engine="openpyxl") as writer:
    churn_summary.to_excel(writer, sheet_name="Churn by Country", index=False)
    high_balance.to_excel(writer, sheet_name="High Balance Churn", index=False)
    age_churn.to_excel(writer, sheet_name="Churn by Age Group", index=False)
    gender_churn.to_excel(writer, sheet_name="Churn by Gender", index=False)

print("Excel file updated with all 4 sheets!")

Excel file updated with all 4 sheets!
